R/stan_betareg.fit.R

```diff
@@ -0,0 +1,225 @@
+    assign(i, prior_ops[[i]])
+  
+  ok_dists <- nlist("normal", student_t = "t", "cauchy", "hs", "hs_plus")
+  ok_intercept_dists <- ok_dists[1:3]
+  
+  
+  # prior distributions (handle_glm_prior() from data_block.R)
+  prior_stuff <- handle_glm_prior(prior, nvars, link, default_scale = 2.5)
+  for (i in names(prior_stuff)) # prior_{dist, mean, scale, df}
+    assign(i, prior_stuff[[i]])
+  prior_intercept_stuff <- handle_glm_prior(prior_intercept, nvars = 1, default_scale = 10,
+                                            link, ok_dists = 
+                                              nlist("normal", student_t = "t", "cauchy"))
+  names(prior_intercept_stuff) <- paste0(names(prior_intercept_stuff), "_for_intercept")
+  for (i in names(prior_intercept_stuff)) # prior_{dist, mean, scale, df}_for_intercept
+    assign(i, prior_intercept_stuff[[i]])
+  
+  # prior distributions for parameters on z variables
+  prior_stuff_z <- handle_glm_prior(prior_z, nvars_z, link = link.phi, default_scale = 2.5)
+  for (i in names(prior_stuff_z))
+    assign(paste0(i,"_z"), prior_stuff_z[[i]])
+  prior_intercept_stuff_z <- handle_glm_prior(prior_intercept_z, nvars = 1, link = link.phi, 
+                                              default_scale = 10,
+                                              ok_dists = nlist("normal", student_t = "t", "cauchy"))
+  names(prior_intercept_stuff_z) <- paste0(names(prior_intercept_stuff_z),"_for_intercept_z")
+  for (i in names(prior_intercept_stuff_z))
+    assign(paste0(i), prior_intercept_stuff_z[[i]])
+
+  if (nvars_z == 0) {
+      prior_mean_z <- double()
+      prior_scale_z <- double()
+      prior_df_z <- integer()
+  }
+
+  # create entries in the data block of the .stan file
+  standata <- nlist(
+    N = nrow(xtemp), K = ncol(xtemp), xbar = as.array(xbar), dense_X = !sparse, # TRUE, sparse = FALSE,
+    X = array(xtemp, dim = c(1L, dim(xtemp))),
+    nnz_X = 0L, 
+    w_X = double(), 
+    v_X = integer(), 
+    u_X = integer(), 
+    y = y, 
+    prior_PD, has_intercept, family = 4L, link = link_num, 
+    prior_dist, prior_mean, prior_scale = as.array(pmin(.Machine$double.xmax, prior_scale)), prior_df,
+    prior_dist_for_intercept, prior_mean_for_intercept = c(prior_mean_for_intercept), 
+    prior_scale_for_intercept = min(.Machine$double.xmax, prior_scale_for_intercept), 
+    prior_df_for_intercept = c(prior_df_for_intercept),
+    prior_scale_for_dispersion = prior_scale_for_dispersion %ORifINF% 0,
+    has_weights = length(weights) > 0, weights = double(),
+    has_offset = length(offset) > 0, offset = double(),
+    t = 0L, 
+    p = integer(), 
+    l = integer(), 
+    q = 0L, 
+    len_theta_L = 0L, shape = double(), scale = double(), 
+    len_concentration = 0L, concentration = double(),
+    len_regularization = 0L, regularization = double(),
+    num_non_zero = 0L, 
+    w = double(), 
+    v = integer(), 
+    u = integer(),
+    z_dim = nvars_z, # ncol(z),
```

Comment:
is this comment indicating that `nvars_z` is equal to `ncol(z)`? if not then it's confusing